Xamarin.Forms.Platform.UAP/MediaElementRenderer.cs

```diff
@@ -0,0 +1,296 @@
+					}
+					break;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Paused:
+				case Windows.UI.Xaml.Media.MediaElementState.Stopped:
+				case Windows.UI.Xaml.Media.MediaElementState.Closed:
+					if (Element.KeepScreenOn)
+					{
+						_request.RequestRelease();
+					}
+					break;
+			}
+
+			Controller.CurrentState = FromWindowsMediaElementState(Control.CurrentState);
+		}
+
+		static MediaElementState FromWindowsMediaElementState(Windows.UI.Xaml.Media.MediaElementState state)
+		{
+			switch(state)
+			{
+				case Windows.UI.Xaml.Media.MediaElementState.Buffering:
+					return MediaElementState.Buffering;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Closed:
+					return MediaElementState.Closed;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Opening:
+					return MediaElementState.Opening;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Paused:
+					return MediaElementState.Paused;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Playing:
+					return MediaElementState.Playing;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Stopped:
+					return MediaElementState.Stopped;
+			}
+
+			throw new ArgumentOutOfRangeException();
+		}
+		
+		void BufferingProgressChanged(DependencyObject sender, DependencyProperty dp)
+		{
+			Controller.BufferingProgress = ((Windows.UI.Xaml.Controls.MediaElement)sender).BufferingProgress;
+		}
+
+		void PositionChanged(DependencyObject sender, DependencyProperty dp)
+		{
+			Controller.Position = ((Windows.UI.Xaml.Controls.MediaElement)sender).Position;
+		}
+
+		void ControlSeekCompleted(object sender, RoutedEventArgs e)
+		{
+			Controller.Position = ((Windows.UI.Xaml.Controls.MediaElement)sender).Position;
+			Controller.OnSeekCompleted();
+		}
+
+		protected override void OnElementPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{	
+				case nameof(MediaElement.Aspect):
+					Control.Stretch = Element.Aspect.ToStretch();
+					break;
+
+				case nameof(MediaElement.AutoPlay):
+					Control.AutoPlay = Element.AutoPlay;
+					break;
+
+				case nameof(MediaElement.IsLooping):
+					Control.IsLooping = Element.IsLooping;
+					break;
+
+				case nameof(MediaElement.KeepScreenOn):
+					if (Element.KeepScreenOn)
+					{
+						if (Control.CurrentState == Windows.UI.Xaml.Media.MediaElementState.Playing)
+						{
+							_request.RequestActive();
+						}
+					}
+					else
+					{
+						_request.RequestRelease();
+					}
+					break;
+
+				case nameof(MediaElement.ShowsPlaybackControls):
+					Control.AreTransportControlsEnabled = Element.ShowsPlaybackControls;
+					break;
+				
+				case nameof(MediaElement.Source):
+					UpdateSource();
+					break;
+
+				case nameof(MediaElement.Width):
+					if (Element.Width > 0)
+					{
+						Width = Element.Width;
+					}
+					break;
+
+				case nameof(MediaElement.Height):
+					if (Element.Height > 0)
+					{
+						Height = Element.Height;
```

Comment:
I think the limit value is better than ignoring set to zero or negative
`Height = Math.Max(0, Element.Height)`